SCons/Tool/ninja/Rules.py

```diff
@@ -37,16 +37,16 @@ def _install_action_function(_env, node):
 def _mkdir_action_function(env, node):
     return {
         "outputs": get_outputs(node),
-        "rule": get_rule(node, "CMD"),
+        "rule": get_rule(node, "GENERATED_CMD"),
         # implicit explicitly omitted, we translate these so they can be
         # used by anything that depends on these but commonly this is
         # hit with a node that will depend on all of the fake
         # srcnode's that SCons will never give us a rule for leading
         # to an invalid ninja file.
         "variables": {
             # On Windows mkdir "-p" is always on
-            "cmd": "{mkdir}".format(
-                mkdir="mkdir $out & exit 0" if env["PLATFORM"] == "win32" else "mkdir -p $out",
+            "cmd": "mkdir {args}".format(
+                args = ' '.join(get_outputs(node)) + " & exit /b 0" if env["PLATFORM"] == "win32" else "-p " + ' '.join(get_outputs(node)),
```

Comment:
the windows bit wasn't working here, because a CMD rule is supposed to a be a singular command, in which the $in and $out are supplied after the command. If you want to have something more complicated, then GENERATED_CMD is more suited, which lets you handle setting the inputs and outputs.